core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -118,6 +120,7 @@
     private volatile AsyncTranslogFSync fsyncTask;
     private final ThreadPool threadPool;
     private final BigArrays bigArrays;
+    private volatile AsyncGlobalCheckpointTask globalCheckpointTask;
```

Comment:
I see, it's scheduled from the constructor of `BaseAsyncTask`. I do not think `volatile` is needed here then, and this field can be marked `final`.